src/components/SearchBar.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, createElement } from "react";
+import { HybridConstraint, ListView, WrapperProps } from "./OfflineSearch";
+
+export interface SearchBarProps extends WrapperProps {
+    listView?: ListView;
+}
+
+interface SearchBarState {
+    buttonVisibility?: string;
+}
+
+export class SearchBar extends Component<SearchBarProps, SearchBarState> {
+    // internal variables
+    private searchButton: HTMLButtonElement;
+    private searchInput: HTMLInputElement;
+
+    constructor(props: SearchBarProps) {
+        super(props);
+
+        this.state = {
+            buttonVisibility: "hidden"
+        };
+        this.onClear = this.onClear.bind(this);
+        this.onSearchKeyDown = this.onSearchKeyDown.bind(this);
+    }
+
+    render() {
+        return this.props.showSearchBar
+            ?
+            createElement("div", { className: "search-container" },
+                createElement("span", { className: "glyphicon glyphicon-search" }),
+                createElement("input", {
+                    className: "form-control", placeholder: "Search",
+                    ref: input => this.searchInput = input as HTMLInputElement
+                }),
+                createElement("button", {
+                    className: `btn-transparent ${this.state.buttonVisibility}`,
+                    ref: button => this.searchButton = button as HTMLButtonElement
+                },
+                    createElement("span", { className: "glyphicon glyphicon-remove" })
+                )
+            )
+            : createElement("div", { className: "search-container-hidden" });
```

Comment:
use `widget-offline-search-bar-hidden` or `widget-offline-search-bar hidden`
use hidden to leverage on the bootstrap styling. hiding it for you!